custom/ucr_ext/expressions.py

```diff
@@ -404,3 +421,127 @@ def get_case_forms_by_date(spec, context):
             }
         }
     return ExpressionFactory.from_spec(spec, context)
+
+
+def get_case_history_by_date(spec, context):
+    GetCaseHistoryByDateSpec.wrap(spec)
+
+    filters = []
+    if spec['start_date'] is not None:
+        start_date_filter = {
+            'operator': 'gte',
+            'expression': {
+                'datatype': 'integer',
+                'from_date_expression': spec['start_date'],
+                'type': 'diff_days',
+                'to_date_expression': {
+                    'datatype': 'date',
+                    'type': 'property_name',
+                    'property_name': 'date'
+                }
+            },
+            'type': 'boolean_expression',
+            'property_value': 0
+        }
+        filters.append(start_date_filter)
+    if spec['end_date'] is not None:
+        end_date_filter = {
+            'operator': 'gte',
+            'expression': {
+                'datatype': 'integer',
+                'from_date_expression': {
+                    'datatype': 'date',
+                    'type': 'property_name',
+                    'property_name': 'date'
+                },
+                'type': 'diff_days',
+                'to_date_expression': spec['end_date']
+            },
+            'type': 'boolean_expression',
+            'property_value': 0
+        }
+        filters.append(end_date_filter)
+    if spec['filter'] is not None:
+        filters.append(spec['filter'])
+
+    if len(filters) > 0:
+        spec = {
+            "type": "sort_items",
+            "sort_expression": {
+                'datatype': 'date',
+                'type': 'property_name',
+                'property_name': 'date'
+            },
+            'items_expression': {
+                'filter_expression': {
+                    'type': 'and',
+                    'filters': filters
+                },
+                'type': 'filter_items',
+                'items_expression': {
+                    'datatype': 'array',
+                    'type': 'ext_root_property_name',
+                    'property_name': 'actions'
+                }
+            }
+        }
+    else:
+        spec = {
+            'type': 'sort_items',
+            'sort_expression': {
+                'datatype': 'date',
+                'type': 'property_name',
+                'property_name': 'date'
+            },
+            'items_expression': {
+                'datatype': 'array',
+                'type': 'ext_root_property_name',
+                'property_name': 'actions'
+            }
+        }
+    return ExpressionFactory.from_spec(spec, context)
+
+
+def get_last_case_property_update(spec, context):
+    GetLastCasePropertyUpdateSpec.wrap(spec)

```

Comment:
and I think it would be possible to write a `get_last_case_property_update_sql` expression - at which point you could remove the assertion. but as @snopoke mentioned it would not be very performant.